src/components/input/_input.scss

```diff
@@ -4,14 +4,22 @@
 
 @include exports("input") {
   .govuk-c-input {
+    @include govuk-responsive-margin($govuk-spacing-responsive-6, "bottom");
+  }
+
+  .govuk-c-input--error {
```

Comment:
Am I right in thinking that we're supporting two behaviours here – _either_ having an input in error state, or having a form-group in error state?

I think we should consider simplifying things such that inputs are always in a form-group, and the left-hand border always comes from the form-group?

Reply:
You could have both the larger form group in error state and the input itself in an error state. Like this:
![screen shot 2018-02-12 at 17 54 44](https://user-images.githubusercontent.com/5007934/36111546-62701dac-101e-11e8-813f-e34dc5d9472f.png)

Or do you mean that the wrapper (`govuk-c-input`) should actually be `govuk-form-group`? To me it feels more semantic to have the wrapper name sound like part of the component markup (like `govuk-c-input` is here), instead of `govuk-form-group` that might get missed out when copy pasting HTML as it sounds like something that you might want to use if you grouping inputs together. Although that's the way it is in Elements 🤔

Reply:
In that screenshot, the `input__input--error` class is being used to make the input border red, but the additional left-hand border is coming from the form-group, not the `input--error` class that this comment is referencing. If that class was added, you'd end up with two left-hand borders in that screenshot.

So… I guess I am basically saying the second thing? The benefit in my mind is that it's simpler to understand (the left-hand border always comes from the fieldset or form-group) and there's less repetition in the code. But as you say, users might find the form-group confusing.

Reply:
Thanks @36degrees. Now everything uses `form-o-group`. More info here: https://github.com/alphagov/govuk-frontend/pull/519#issuecomment-365544439